### DISCUSSION TITLE
Does middleware run in dev mode?

### DISCUSSION THREAD

**Discussion by vedantroy** [vercel/next.js#50880] [Help] [answered]
### Summary

I have middleware:

![image](https://github.com/vercel/next.js/assets/8407819/7e49d924-9568-4272-835c-8ae8f8d3c906)

but it doesn't seem to run when doing `npm run dev`

**Comment by icyJoseph** [comment, marked as answer]
It does run in dev mode, it even runs side by side your app, when you do `next start`, it is only on Vercel, or other serverless+edge vendors, that the middleware is separated from the server.

Perhaps you are using custom page extensions? Do you have matchers inside the middleware?

**Reply by vedantroy** [reply]
Not really sure what's going on, it runs with `npm run start`, so--maybe it is also running in dev mode, and I don't notice because HTTP BASIC AUTH is cached or something like that.

**Comment by PSoltes** [comment]
I have a related question. I have an react app that I deploy to vercel. In that react app I'd like to use a middleware. When I deploy that app to vercel middleware runs fine. Is there a way to replicate this locally? I've tried running `vercel dev` as it states that it [should replicate deployment locally](https://vercel.com/docs/cli/dev) thus giving ability to test rewrites, middleware and so on, but no luck.
